core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -70,6 +70,36 @@ public Table loadTable(TableIdentifier ident) {
     return tableCache.get(canonicalizeIdentifier(ident), catalog::loadTable);
   }
 
+  @Override
+  public boolean createNamespace(Namespace namespace) {
+    return false;
+  }
+
+  @Override
+  public List<Namespace> listNamespaces() {
+    return null;
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    return null;
+  }
+
+  @Override
+  public Map<String, String>  loadNamespaceMetadata(Namespace namespace) {
+    return null;
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    return false;
+  }
+
+  @Override
+  public boolean alterNamespace(Namespace namespace) {
+    return false;
+  }
```

Comment:
The implementation above is not correct, we should delegate namespace operations to the actual catalog, or we could cache them.

Reply:
fixed: delete from CachingCatalog